ddtrace/internal/constants.py

```diff
@@ -1,9 +1,12 @@
+from ddtrace.internal.serverless import in_gcp_function
+
+
 PROPAGATION_STYLE_DATADOG = "datadog"
 PROPAGATION_STYLE_B3 = "b3multi"
 PROPAGATION_STYLE_B3_SINGLE_HEADER = "b3 single header"
 _PROPAGATION_STYLE_W3C_TRACECONTEXT = "tracecontext"
 _PROPAGATION_STYLE_NONE = "none"
-_PROPAGATION_STYLE_DEFAULT = "tracecontext,datadog"
+_PROPAGATION_STYLE_DEFAULT = "datadog,tracecontext" if in_gcp_function() else "tracecontext,datadog"
```

Comment:
with google cloud functions, requests can include both tracecontext & datadog headers. this is because google cloud functions will always inject a `tracecontext`, even if a function is directly invoked.

In the even that a function is directly invoked by another datadog instrumented service upstream, we want datadog headers to be extracted (not tracecontext), so try to extract datadog headers first.

Is there currently any reason we prioritize tracecontext over datadog propagation?

Reply:
removed